src/audit_log/audit_log.cc

```diff
@@ -372,6 +373,23 @@ bool AuditLog::merge(AuditLog *from, std::string *error) {
     return init(error);
 }
 
+bool AuditLog::reopen(std::string *error) {
+    bool success = true;
+
+    // NOTE: We could delegate this responsibility to the writer interface,
+    // but today, since the Serial and Parallel writers actually reference
+    // the log paths maintained by this instance, we might as well go the
+    // simpler path and handle reopen here.
+    if (!m_path1.empty()) {
+        success = utils::SharedFiles::getInstance().reopen(m_path1, error);
+    }
+    if (success && !m_path2.empty()) {
```

Comment:
I'd considered that but believe I made the wrong choice. :) I agree with you and will fix that.

Reply:
I think one reason I avoided it is to avoid potentially dealing with two different error strings. It may make sense to concatenate or otherwise combine them.